File: port_java/com/boare/cadencii/CadenciiCommand.java
/*
 * CadenciiCommand.cs
 * Copyright (c) 2008-2009 kbinani
 *
 * This file is part of Boare.Cadencii.
 *
 * Boare.Cadencii is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * Boare.Cadencii is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
package com.boare.cadencii;

import java.util.*;

import com.boare.vsq.*;

/// <summary>
/// Undo/Redoを実現するためのコマンド。
/// Boare.Lib.Vsq.VsqFileレベルのコマンドは、Type=VsqCommandとして取り扱う。
/// Boare.Cadencii.VsqFileExレベルのコマンドは、Argsに処理内容を格納して取り扱う。
/// </summary>
public class CadenciiCommand implements Command {
    public CadenciiCommandType type;
    public VsqCommand vsqCommand;
    public Command m_parent;
    public Object[] args;
    private Vector<Command> m_child = new Vector<Command>();

    public CadenciiCommand( VsqCommand command ) {
        type = CadenciiCommandType.VsqCommand;
        vsqCommand = command;
        m_child = new Vector<Command>();
    }

    public CadenciiCommand() {
        m_child = new Vector<Command>();
    }

    public Command getParent(){
        return m_parent;
    }

    public void setParent( Command value ){
        m_parent = value;
    }

    public Vector<Command> getChild(){
        return m_child;
    }
}

File: src/Cadencii/CadenciiCommand.cs
/*
 * CadenciiCommand.cs
 * Copyright © 2008-2011 kbinani
 *
 * This file is part of cadencii.
 *
 * cadencii is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * cadencii is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
using System;
using System.Collections.Generic;
using cadencii.vsq;
using cadencii.java.util;

namespace cadencii
{

    /// <summary>
    /// Undo/Redoを実現するためのコマンド。
    /// Boare.Lib.Vsq.VsqFileレベルのコマンドは、Type=VsqCommandとして取り扱う。
    /// Boare.Cadencii.VsqFileExレベルのコマンドは、Argsに処理内容を格納して取り扱う。
    /// </summary>
    [Serializable]
    public class CadenciiCommand : ICommand
    {
        public CadenciiCommandType type;
        public VsqCommand vsqCommand;
        private ICommand mParent;
        public Object[] args;
        private List<ICommand> mChild = new List<ICommand>();

        public CadenciiCommand(VsqCommand command)
        {
            type = CadenciiCommandType.VSQ_COMMAND;
            vsqCommand = command;
            mChild = new List<ICommand>();
        }

        public CadenciiCommand()
        {
            mChild = new List<ICommand>();
        }

        public ICommand getParent()
        {
            return mParent;
        }

        public void setParent(ICommand value)
        {
            mParent = value;
        }

        public List<ICommand> getChild()
        {
            return mChild;
        }
    }

}
